src/utils/promise.js

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2019 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This is only here to allow access to methods like done for the time being
+import Promise from "bluebird";
+
+// @flow
+
+// Returns a promise which resolves with a given value after the given number of ms
+export const sleep = (ms: number, value: any): Promise => new Promise((resolve => { setTimeout(resolve, ms, value); }));
+
+// Returns a promise which resolves when the input promise resolves with its value
+// or when the timeout of ms is reached with the value of given timeoutValue
+export async function timeout(promise: Promise, timeoutValue: any, ms: number): Promise {
+    const timeoutPromise = new Promise((resolve) => {
+        const timeoutId = setTimeout(resolve, ms, timeoutValue);
+        promise.then(() => {
+            clearTimeout(timeoutId);
+        });
+    });
+
+    return Promise.race([promise, timeoutPromise]);
+}
```

Comment:
I can't see where this is used?

Reply:
True, it isn't yet, but will be in the next PR, want me to split it out?